src/com/biocollapse/service/SimulationService.java

```diff
@@ -1,5 +1,55 @@
 package src.com.biocollapse.service;
 
+import src.com.biocollapse.model.Hospital;
+import src.com.biocollapse.model.Human;
+import src.com.biocollapse.model.LiveStatistics;
+
 public class SimulationService {
-  
+
+    public static LiveStatistics calculateLiveStatistics(Human[] humans, Hospital[] hospitals) {
+        int infectedCounter = 0;
+        int healthyCounter = humans.length;
+        int immuneCounter = 0;
+        int aliveCounter = humans.length;
+        int deathCounter = 0;
+        int hospitalCapacity = 0;
+        int usedHospitalCapacity = 0;
+        double hospitalCapacityRatio = 0.0;
+
+        for (Human human : humans) {
+            if (human.isInfected()) {
+                infectedCounter++;
+                healthyCounter = (healthyCounter > 0) ? healthyCounter -= 1 : healthyCounter;
+            }
+            if (!human.isAlive()) {
+                deathCounter++;
+                aliveCounter = (aliveCounter > 0) ? aliveCounter -= 1 : aliveCounter;
+                infectedCounter = (infectedCounter > 0) ? infectedCounter -= 1 : infectedCounter;
+            }
+            // This case can only occur when someone was infected is healthy again
+            if (human.isImmune()) {
+                immuneCounter++;
+                healthyCounter++;
+            }
+        }
+
+        for (Hospital hospital : hospitals) {
+            hospitalCapacity += hospital.getCapacity();
+            usedHospitalCapacity += hospital.getUsedCapacity();
+        }
+
+        if (hospitalCapacity > 0) {
+            hospitalCapacityRatio = usedHospitalCapacity / hospitalCapacity;
+        } else {
+            // Idea: Instead of returning 0, when we got an Error, we could return Pi,
+            // because Pi can never be our result in the calculation, we know that when it
+            // is PI it has to be an error.
+            hospitalCapacityRatio = Math.PI;
```

Comment:
hmm. I would not start using random numbers as a default state. Why don't you make hospitalCapacityRatio an "Integer" instead of int, which can be null?

If we have an error, we should definently log it for debugging purposes ASAP.

Reply:
You are right. I changed the type from double to Integer and return -1 as an Error now + log it in Console for debugging. This way the frontend can check if the value is < 0 then it is an error in this tick.

I should add, that i cannot think of a scenario where we are not able to calculate the sum of the hospitalCapacity i just added it just to be sure